client/ayon_traypublisher/plugins/create/create_editorial_advanced.py

```diff
@@ -0,0 +1,1341 @@
+                self._get_clip_instances(
+                    folder_entity,
+                    otio_timeline,
+                    media_folder_path,
+                    clip_instance_properties,
+                    allowed_product_type_presets,
+                    sequence_name,
+                    ignore_clip_no_content,
+                )
+
+                # create otio editorial instance
+                self._create_otio_instance(
+                    product_name, instance_data, sequence_path, otio_timeline
+                )
+
+    def _create_otio_instance(
+        self, product_name, instance_data, sequence_path, otio_timeline
+    ):
+        """Otio instance creating function
+
+        Args:
+            product_name (str): Product name.
+            data (dict): instance data
+            sequence_path (str): path to sequence file
+            otio_timeline (otio.Timeline): otio timeline object
+        """
+        # Pass precreate data to creator attributes
+        instance_data.update(
+            {
+                "sequenceFilePath": sequence_path,
+                "otioTimeline": otio.adapters.write_to_string(otio_timeline),
+            }
+        )
+        new_instance = CreatedInstance(
+            self.product_type, product_name, instance_data, self
+        )
+        self._store_new_instance(new_instance)
+
+    def _create_otio_timeline(self, sequence_path, fps):
+        """Creating otio timeline from sequence path
+
+        Args:
+            sequence_path (str): path to sequence file
+            fps (float): frame per second
+
+        Returns:
+            otio.Timeline: otio timeline object
+        """
+        # get editorial sequence file into otio timeline object
+        extension = os.path.splitext(sequence_path)[1]
+
+        kwargs = {}
+        if extension == ".edl":
+            # EDL has no frame rate embedded so needs explicit
+            # frame rate else 24 is assumed.
+            kwargs["rate"] = fps
+            kwargs["ignore_timecode_mismatch"] = True
+
+        return otio.adapters.read_from_file(sequence_path, **kwargs)
+
+    def _get_path_from_file_data(self, file_path_data, multi=False):
+        """Converting creator path data to single path string
+
+        Args:
+            file_path_data (FileDefItem): creator path data inputs
+            multi (bool): switch to multiple files mode
+
+        Raises:
+            FileExistsError: in case nothing had been set
+
+        Returns:
+            str: path string
+        """
+        return_path_list = []
+
+        if isinstance(file_path_data, list):
+            return_path_list = [
+                os.path.join(f["directory"], f["filenames"][0])
+                for f in file_path_data
+            ]
+
+        if not return_path_list:
+            raise FileExistsError(
+                f"File path was not added: {file_path_data}")
+
+        return return_path_list if multi else return_path_list[0]
```

Comment:
I would not use `return` or `list` in variable name. If the input is from FileDef attribute then you should always get list and you should also iterate over filenames. Also raising `FileExistsError` is not ideal, it probably should be `CreatorError` or something else, might even worth to not raise anything and let the calling code to raise error if needed with specific message (Raise `CreatorError` with `Path to .edl file is not filled, please make sure it is filled to proceed.`)?
```suggestion
        output_paths = []
        for item in file_path_data:
            dirpath = item["directory"]
            for filename in item["filenames"]:
                output_paths.append(os.path.join(dirpath, filename)

        if not output_paths:
            raise CreatorError(
                f"File path was not added: {file_path_data}")

        return output_paths if multi else output_paths[0]
```